### PR TITLE
Add deployment types for cloud

### PR DESCRIPTION
**Pull request by harsh-px**
Screenshots below

Screen Shot 2019-04-02 at 11 24 57 AM
Screen Shot 2019-04-02 at 11 25 07 AM

Signed-off-by: Harsh Desai <harsh@portworx.com>

### CODE REVIEW

File: content/cloud-references/deployment-arch/_index.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+linkTitle: Deployment architectures
+description: Understand various deployment architectures for Portworx on Cloud
+keywords: cloud,asg,aws,gcp,vsphere,disks
+weight: 1
+series: cloud-references
+---
+
+In cloud environments, Portworx can manage the entire lifecycle of your storage needs. The includes:
+
+* Managing the volumes exposed to your application containers
```
Comment (gourao) [inline content/cloud-references/deployment-arch/_index.md]: I would remove these 4 lines... they describe the ASG mode and not really what this particular article is about

File: content/cloud-references/deployment-arch/_index.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
+---
+title: Deployment architectures for Portworx on Cloud
+linkTitle: Deployment architectures
+description: Understand various deployment architectures for Portworx on Cloud
+keywords: cloud,asg,aws,gcp,vsphere,disks
+weight: 1
+series: cloud-references
+---
+
+In cloud environments, Portworx can manage the entire lifecycle of your storage needs. The includes:
```
Comment (gourao) [inline content/cloud-references/deployment-arch/_index.md]: I would take this out... its about ASG and not really the primary topic of this article (deployment models)

File: content/cloud-references/deployment-arch/_index.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+* Managing the volumes exposed to your application containers
+* The backend disks on the cloud
+* And everything in between
+
+There are 2 primary approaches you can take when architecting your Portworx deployment on cloud.
```
Comment (gourao) [inline content/cloud-references/deployment-arch/_index.md]: There are 2 primary approaches you can take when architecting your Portworx deployment on cloud:
a) Disaggregated where the storage nodes are separate from the compute nodes
b) Converged where compute and storage nodes are the same

These two approaches are discussed below.

File: content/cloud-references/deployment-arch/_index.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+### What
+
+* Separate Storage cluster (green above). Nodes in this cluster have disks.
+* Separate Compute cluster (yellow above). The application container workloads run in this cluster.
+* The Portworx installation (orange above) spans across both of these clusters to provide a single storage fabric.
+* The Storage and Compute clusters are different orchestrator clusters (e.g different Kubernetes clusters) that share the storage fabric.
+
+### Why
+
+Compute clusters on cloud are usually very dynamic and constantly changing due to a variety of reasons:
```
Comment (gourao) [inline content/cloud-references/deployment-arch/_index.md]: You would chose this option if you have a very dynamic compute environment, where the number of compute nodes can elastically increase or decrease based on workload demand.  Some examples of what can cause this elasticity are:

File: content/cloud-references/deployment-arch/_index.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+* Autoscale up or down due to increasing and decreasing demands. Simple examples of these would be the Autoscaling instance group increasing in size from 3 to 5 instances to handle increasing container workloads.
```
Comment (gourao) [inline content/cloud-references/deployment-arch/_index.md]: "Autoscaling up or down due to increasing and decreasing demands. An example would be to temporarily increase the number of worker nodes from 30 to 50 to handle the number of PODs in the system"

File: content/cloud-references/deployment-arch/_index.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+* Instance upgrades due to kernel updates, security patches etc
+* Orchestrator upgrades (e.g Kubernetes upgrade)
+
+Separating storage and compute clusters mean such scaling & management operations on the storage cluster don't interfere with the compute cluster. And vice versa.
```
Comment (gourao) [inline content/cloud-references/deployment-arch/_index.md]: Separating storage and compute clusters mean such scaling & management operations on the storage cluster don't interfere with the compute cluster, and vice versa.

File: content/cloud-references/deployment-arch/_index.md (hunk 2 above)
Comment (harsh-px) [reply content/cloud-references/deployment-arch/_index.md]: Removed. Now just starting with "There are 2 primary...."

File: content/cloud-references/deployment-arch/_index.md (hunk 1 above)
Comment (harsh-px) [reply content/cloud-references/deployment-arch/_index.md]: removed

File: content/cloud-references/deployment-arch/_index.md (hunk 3 above)
Comment (harsh-px) [reply content/cloud-references/deployment-arch/_index.md]: "Disaggregated" might be a very advanced term for the average reader. I've incorporated your wording for now. We can change it later.

File: content/cloud-references/deployment-arch/_index.md (hunk 4 above)
Comment (harsh-px) [reply content/cloud-references/deployment-arch/_index.md]: reworded

File: content/cloud-references/deployment-arch/_index.md (hunk 5 above)
Comment (harsh-px) [reply content/cloud-references/deployment-arch/_index.md]: done

**Pull request merged**